ChatSpace　DB設計

ryoryo-rk opened the pull request:
WHAT Readme.mdにDB設計、テーブルの内容とアソシエーションを記述。 WHY
Chatspace実装前に必要なテーブルやカラム、テーブル同士の関係性を明確にするため。

ryoryo-rk committed:
ChatSpace　DB設計

WHAT Readme.mdにDB設計、テーブルの内容とアソシエーションを記述。 WHY
Chatspace実装前に必要なテーブルやカラム、テーブル同士の関係性を明確にするため。

TOMARUMARU commented on README.md:
```diff
@@ -23,3 +23,40 @@ Things you may want to cover:
 
 * ...
 
+## user_groupテーブル
```
users_groupsにしましょう！

TOMARUMARU commented on README.md:
```diff
@@ -23,3 +23,40 @@ Things you may want to cover:
+|column|Type|Options|
+|------|----|-------|
+|name|string|null: false, add_index :users, :name, unique: true|
+|email|string|null: false, add_index :users, :email, unique: true|
```
emailにインデックスを設定する必要はございません！

TOMARUMARU commented on README.md:
```diff
@@ -23,3 +23,40 @@ Things you may want to cover:
+
+|column|Type|Options|
+|------|----|-------|
+|name|string|null: false, add_index :users, :name, unique: true|
```
add_index :users
ではなく、
index :true
にしましょう！

TOMARUMARU reviewed (commented):
groupsテーブルが見当たりません。。。
確認お願いいたします！

ryoryo-rk committed:
[README.md]  improvement1

user_group table  rename  'users_groupテーブル'
users_group table  changed  'add_index :users  to  index:true'
add  groupsテーブル

miyagit reviewed (commented):
まずRead meの見た目がおかしくなっているのでその部分を直していただいても宜しいでしょうか？

ryoryo-rk committed:
[README] changed view

README.mdの見た目の改善。

miyagit commented on README.md:
```diff
@@ -23,3 +23,49 @@ Things you may want to cover:
+
+|column|Type|Options|
+|------|----|-------|
+|group_name|string|null: false|
```
こちらnameで大丈夫です。

miyagit commented on README.md:
```diff
@@ -23,3 +23,49 @@ Things you may want to cover:
+|group_name|string|null: false|
+
+### Assosiation
+- has_many :messages
```
アソシエーションはこれだけで宜しいでしょうか？
中間テーブルとの関連性について調べて頂きたく思います。

miyagit commented on README.md:
```diff
@@ -23,3 +23,49 @@ Things you may want to cover:
 
 * ...
```
Readmeのデフォルトの表記は消していただいても大丈夫です。

ryoryo-rk committed:
README.md

usersテーブルにアソシエーションを追記。
groupsテーブルにアソシエーションを追記。
group_nameをnameに変える。

ryoryo-rk committed:
Add readme

ryoryo-rk committed:
Change README.md

不要なアソシエーションを削除。

TOMARUMARU commented on README.md:
```diff
@@ -1,25 +1,51 @@
 
-* System dependencies
+### Association
+- belongs_to :groups
```
groupsではなくgroupですね！

TOMARUMARU commented on README.md:
```diff
@@ -1,25 +1,51 @@
+
+### Assosiation
+-has_meny :messages
+-has_many :user_group, through: :users_group
```
アソシエーションがこれだとしっかり組めていません！

TOMARUMARU commented on README.md:
```diff
@@ -1,25 +1,51 @@
+
+### Assosiation
+-has_many :messages
+-has_many :users, through: :users_group
```
アソシエーションがこれだとしっかり組めていません！

TOMARUMARU commented on README.md:
```diff
@@ -1,25 +1,51 @@
+
+|column|Type|Options|
+|------|----|-------|
+|name|string|null: false|
```
一意性制約をかけましょう。

TOMARUMARU reviewed (commented):
キー制約についてもう一度よく調べてください。
どのカラムにどの制約をつけるのかもう一度よく考えてみてください。

ryoryo-rk committed:
Changed README.md

usersテーブルのアソシエーションの変更。
groupsテーブルにnameのカラムに一意性制約を追加。
mesageテーブルのアソシエーションの名前を改善。

ryoryo-rk committed:
README.md

アソシエーションの改善。
インデックスを追記。
